speech/cloud-client/src/main/java/com/example/speech/Recognize.java

```diff
@@ -779,4 +797,363 @@ public static void transcribeFileWithMetadata(String fileName) throws Exception
     }
   }
   // [END speech_transcribe_file_with_metadata]
+
+  // [START speech_transcribe_diarization]
+  /**
+   * Transcribe the given audio file using speaker diarization.
+   *
+   * @param fileName the path to an audio file.
+   */
+  public static void transcribeDiarization(String fileName) throws Exception {
+    Path path = Paths.get(fileName);
+    byte[] content = Files.readAllBytes(path);
+
+    try (SpeechClient speechClient = SpeechClient.create()) {
+      // Get the contents of the local audio file
+      RecognitionAudio recognitionAudio =
+          RecognitionAudio.newBuilder().setContent(ByteString.copyFrom(content)).build();
+
+      // Configure request to enable enhanced models
+      RecognitionConfig config =
+          RecognitionConfig.newBuilder()
+              .setEncoding(AudioEncoding.LINEAR16)
+              .setLanguageCode("en-US")
+              .setSampleRateHertz(8000)
+              .setEnableSpeakerDiarization(true)
+              .setDiarizationSpeakerCount(2)
+              .setEnableAutomaticPunctuation(true)
```

Comment:
Let's remove the automatic punctuation for this sample. Unless its needed.

Reply:
+1 Especially in a sample, it should only have the minimum needed to get the feature you're demonstrating working.